### ISSUE TITLE
Problems building core/filter on CentOS7 after https://github.com/ufo-kit/ufo-core/pull/169 (cpp)

### ISSUE THREAD

**Issue by sgasilov** [ufo-kit/ufo-core#180]
Hello,

FYI, build fails because of cpp in tests (see attached files). 
[build-ufocore.log](https://github.com/ufo-kit/ufo-core/files/8641887/build-ufocore.log)

We ended up using -DWITH_TESTS=OFF when calling cmake. 

Suggestion: if these new feature is truly important and you want to keep it, could you change makefile so that it doesn't compile "tests" by default? That will make things simpler for ordinary people who try to deploy ufo and do not need those tests anyways. Expert users who need the tests will know how to enable that.

**Comment by tfarago** [comment]
@MarcusZuber, can you check this, it's about the max_input_nodes?

**Comment by MarcusZuber** [comment]
Sure.

**Comment by MarcusZuber** [comment]
Ok, that was an easy one.
The tests require c++11 which I didn't add as requirement to CMake and meson.

I will do these changes:

- [ ] Make tests an optional target
- [ ] Add c++11 as a requirement for the tests

**Comment by MarcusZuber** [comment]
Hi Sergey,
could you test the newest version?
Now the tests are not build by default.

Could you also try to build the tests? I'm curious how cmake and meson reacts if you compiler does not implement c++11.

**Comment by ldorofeeva** [comment]
Hey Marcus,

I've tested it on centos7.

The default build works ok, test are not being built as expected.

If I attempt to build the tests, I get the following:
```
[dorofel@localhost ufo-core]$ cmake . -DWITH_TESTS=ON
-- checking for module 'python3-embed'
--   package 'python3-embed' not found
-- checking for module 'python3'
--   package 'python3' not found
-- checking for module 'python'
--   package 'python' not found
-- checking for module 'gtk-doc'
--   package 'gtk-doc' not found
-- Configuring done
-- Generating done
-- Build files have been written to: /staff/dorofel/Projects/ufo-core

[dorofel@localhost ufo-core]$ make all
[ 72%] Built target ufo
[ 76%] Built target gir
[ 79%] Built target manual
[ 81%] Built target ufo-launch
[ 83%] Built target ufo-query
[ 86%] Built target ufo-runjson
[ 88%] Building CXX object tests/unit/CMakeFiles/test-suite.dir/test-max-input-nodes.cpp.o
In file included from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:21:0:
/staff/dorofel/Projects/ufo-core/config.h:7:0: warning: "GLIB_VERSION_MIN_REQUIRED" redefined [enabled by default]
 #define GLIB_VERSION_MIN_REQUIRED   GLIB_VERSION_2_38
 ^
In file included from /usr/include/glib-2.0/glib/gtypes.h:34:0,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:20:
/usr/include/glib-2.0/glib/gversionmacros.h:237:0: note: this is the location of the previous definition
 # define GLIB_VERSION_MIN_REQUIRED      (GLIB_VERSION_CUR_STABLE)
 ^
In file included from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:21:0:
/staff/dorofel/Projects/ufo-core/config.h:8:0: warning: "GLIB_VERSION_MAX_ALLOWED" redefined [enabled by default]
 #define GLIB_VERSION_MAX_ALLOWED    GLIB_VERSION_2_38
 ^
In file included from /usr/include/glib-2.0/glib/gtypes.h:34:0,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:20:
/usr/include/glib-2.0/glib/gversionmacros.h:266:0: note: this is the location of the previous definition
 # define GLIB_VERSION_MAX_ALLOWED      (GLIB_VERSION_CUR_STABLE)                                                                                                                                                                            
 ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp: In function ‘std::string build_kernel(unsigned int)’:
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:43:34: error: ‘sprintf’ was not declared in this scope
         sprintf(number, "%06i", i);                                                                                                                                                                                                         
                                  ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:53:34: error: ‘sprintf’ was not declared in this scope
         sprintf(number, "%06i", i);                                                                                                                                                                                                         
                                  ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp: In function ‘void test_n_inputs(unsigned int)’:
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:87:10: error: ‘opencl_kernel’ does not name a type
     auto opencl_kernel = ufo_plugin_manager_get_task(manager, "opencl", nullptr);                                                                                                                                                           
          ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:88:9: error: ‘opencl_kernel’ was not declared in this scope
     if (opencl_kernel == nullptr)                                                                                                                                                                                                           
         ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:88:26: error: ‘nullptr’ was not declared in this scope
     if (opencl_kernel == nullptr)
                          ^
In file included from /usr/include/glib-2.0/gobject/gobject.h:24:0,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo-basic-ops.h:27,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo.h:25,
                 from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:22:
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:90:29: error: ‘opencl_kernel’ was not declared in this scope
     g_object_set (G_OBJECT (opencl_kernel),
                             ^
/usr/include/glib-2.0/gobject/gtype.h:2224:57: note: in definition of macro ‘_G_TYPE_CIC’
     ((ct*) g_type_check_instance_cast ((GTypeInstance*) ip, gt))
                                                         ^
/usr/include/glib-2.0/gobject/gobject.h:51:38: note: in expansion of macro ‘G_TYPE_CHECK_INSTANCE_CAST’
 #define G_OBJECT(object)            (G_TYPE_CHECK_INSTANCE_CAST ((object), G_TYPE_OBJECT, GObject))
                                      ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:90:19: note: in expansion of macro ‘G_OBJECT’
     g_object_set (G_OBJECT (opencl_kernel),
                   ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:95:10: error: ‘sink’ does not name a type
     auto sink = ufo_plugin_manager_get_task(manager, "null", nullptr);
          ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:96:9: error: ‘sink’ was not declared in this scope
     if (sink == nullptr)
         ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:96:17: error: ‘nullptr’ was not declared in this scope
     if (sink == nullptr)
                 ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:102:16: error: ISO C++ forbids declaration of ‘reader’ with no type [-fpermissive]
     for (auto& reader: readers) {
                ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:102:24: error: range-based ‘for’ loops are not allowed in C++98 mode
     for (auto& reader: readers) {
                        ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:103:69: error: ‘nullptr’ was not declared in this scope
         reader = ufo_plugin_manager_get_task(manager, "dummy-data", nullptr);
                                                                     ^
In file included from /usr/include/glib-2.0/gobject/gobject.h:24:0,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo-basic-ops.h:27,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo.h:25,
                 from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:22:
/usr/include/glib-2.0/gobject/gtype.h:482:88: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
 #define G_TYPE_CHECK_INSTANCE_CAST(instance, g_type, c_type)    (_G_TYPE_CIC ((instance), (g_type), c_type))
                                                                                        ^
/usr/include/glib-2.0/gobject/gtype.h:2224:57: note: in definition of macro ‘_G_TYPE_CIC’
     ((ct*) g_type_check_instance_cast ((GTypeInstance*) ip, gt))
                                                         ^
/usr/include/glib-2.0/gobject/gobject.h:51:38: note: in expansion of macro ‘G_TYPE_CHECK_INSTANCE_CAST’
 #define G_OBJECT(object)            (G_TYPE_CHECK_INSTANCE_CAST ((object), G_TYPE_OBJECT, GObject))
                                      ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:106:23: note: in expansion of macro ‘G_OBJECT’
         g_object_set (G_OBJECT (reader),
                       ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:115:56: error: ‘sink’ was not declared in this scope
     ufo_task_graph_connect_nodes(graph, opencl_kernel, sink);
                                                        ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:116:46: error: ‘nullptr’ was not declared in this scope
     ufo_base_scheduler_run(scheduler, graph, nullptr);
                                              ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:119:22: error: ISO C++ forbids declaration of ‘reader’ with no type [-fpermissive]
     for (const auto& reader: readers) {
                      ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:119:30: error: range-based ‘for’ loops are not allowed in C++98 mode
     for (const auto& reader: readers) {
                              ^
/staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:120:30: error: invalid conversion from ‘int’ to ‘gpointer {aka void*}’ [-fpermissive]
         g_object_unref(reader);
                              ^
In file included from /usr/include/glib-2.0/gobject/gbinding.h:29:0,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo-basic-ops.h:27,
                 from /staff/dorofel/Projects/ufo-core/ufo/ufo.h:25,
                 from /staff/dorofel/Projects/ufo-core/tests/unit/test-max-input-nodes.cpp:22:
/usr/include/glib-2.0/gobject/gobject.h:494:13: error:   initializing argument 1 of ‘void g_object_unref(gpointer)’ [-fpermissive]
 void        g_object_unref                    (gpointer        object);
             ^
make[2]: *** [tests/unit/CMakeFiles/test-suite.dir/test-max-input-nodes.cpp.o] Error 1
make[1]: *** [tests/unit/CMakeFiles/test-suite.dir/all] Error 2
make: *** [all] Error 2

```
